### ISSUE TITLE
Discuss: add JSON Merge Patch (RFC 7396)?

### ISSUE THREAD

**Issue by nlohmann** [nlohmann/json#877]
SQLite's json1 extension (https://www.sqlite.org/json1.html) supports JSON Merge Patch (https://tools.ietf.org/html/rfc7396). As the implementation is trivial and we already support JSON Patch, I think this could be a nice extension to the library.

Example:

```cpp
#include <iostream>
#include "json.hpp"
#include <iomanip> // for std::setw

using json = nlohmann::json;

int main()
{
    // the original document
    json document = R"({
                "title": "Goodbye!",
                "author": {
                    "givenName": "John",
                    "familyName": "Doe"
                },
                "tags": [
                    "example",
                    "sample"
                ],
                "content": "This will be unchanged"
            })"_json;

    // the patch
    json patch = R"({
                "title": "Hello!",
                "phoneNumber": "+01-123-456-7890",
                "author": {
                    "familyName": null
                },
                "tags": [
                    "example"
                ]
            })"_json;

    // apply the patch
    document.merge_patch(patch);

    // output original and patched document
    std::cout << std::setw(4) << document << std::endl;
}
```

output:

```js

{
    "author": {
        "givenName": "John"
    },
    "content": "This will be unchanged",
    "phoneNumber": "+01-123-456-7890",
    "tags": [
        "example"
    ],
    "title": "Hello!"
}
```

See PR #876 for more details.

## Question

- What do you think?
- Any comments on the API, etc.?

**Comment by nlohmann** [comment]
Any comments?

**Comment by nlohmann** [comment]
Merged with #876.
